src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2722,6 +2721,44 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                     x.base.base.loc);
             }
 
+        } else if (AST::is_a<AST::Name_t>(*x.m_iter)) {
+            std::string loop_src_var_name = AST::down_cast<AST::Name_t>(x.m_iter)->m_id;
+            auto loop_src_var_symbol = current_scope->get_symbol(loop_src_var_name);
+            auto loop_src_var_ttype = ASRUtils::symbol_type(loop_src_var_symbol);
+            if( loop_src_var_ttype->type == ASR::ttypeType::Character) {
+                auto int_type = ASR::make_Integer_t(al, x.base.base.loc, 4, nullptr, 0);
+
+                {
+                    // create a new variable called/named __explicit_iterator of type i32 and add it to symbol table
+                    std::string explicit_iter_name = current_scope->get_unique_name("__explicit_iterator");
+                    auto explicit_iter_variable = ASR::make_Variable_t(al, x.base.base.loc, current_scope,
+                        s2c(al, explicit_iter_name), ASR::intentType::Local, nullptr, nullptr, ASR::storage_typeType::Default,
+                        ASRUtils::TYPE(int_type), ASR::abiType::Source, ASR::accessType::Public, ASR::presenceType::Required, false
+                    );
+
+                    current_scope->add_symbol(explicit_iter_name, ASR::down_cast<ASR::symbol_t>(explicit_iter_variable));
+                }
+
+                {
+                    // make loop_end = len(loop_src_var), where loop_src_var is the variable over which
+                    // we are iterating the for in loop
+                    auto loop_src_var = ASR::make_Var_t(al, x.base.base.loc, loop_src_var_symbol);
+                    auto call_str_len = ASR::make_StringLen_t(al, x.base.base.loc, ASRUtils::EXPR(loop_src_var), ASRUtils::TYPE(int_type), nullptr);
+                    loop_end = ASRUtils::EXPR(call_str_len);
+                }
+
+                is_explicit_iterator_required = true;
+            } else if (loop_src_var_ttype->type == ASR::ttypeType::List) {
+                throw SemanticError("Iterating on Lists using for in loop not yet supported as "
+                    "visit_Len() is not yet supported on the LLVM Backend", x.base.base.loc);
```

Comment:
```suggestion
                    "visit_Len() is not yet supported in the LLVM Backend", x.base.base.loc);
```

Reply:
I just pushed a commit with this change.